bliss/base_detect/base_common.py

```diff
@@ -0,0 +1,182 @@
+import warnings
+
+import torch
+
+
+def autopad(k, p=None, d=1):  # kernel, padding, dilation
+    # Pad to 'same' shape outputs
+    if d > 1:
+        k = (
+            d * (k - 1) + 1 if isinstance(k, int) else [d * (x - 1) + 1 for x in k]
+        )  # actual kernel-size
+    if p is None:
+        p = k // 2 if isinstance(k, int) else [x // 2 for x in k]  # auto-pad
+    return p
+
```

Comment:
Seems like there must be a better way than including all this yolo code in our repo. Is it possible to mock/monkeypatch the yolo module, to somehow make it use our `Conv3d` class?

Reply:
Or can we get away with only including a customized version of `parse_model` in our repo, rather than all this yolo code?

Reply:
I think overwriting the yolo `parse_model` could be a promising idea - I was planning on rewriting that function a decent amount anyways since it's a little difficult to inject a dimension `squeeze` layer given how it is currently written (i.e. going from the 5-D 'head' back to 4-D). I don't think we'd have to change anything else besides that and including our own `Conv3d` class. 

Cleaning/handling all of the yolo code ourselves is definitely a lot (as shown by the several `#noqa` and `pylint: disable` statements).

Reply:
I think it's a good idea to get the 3D conv working end-to-end with the extra YOLO code first, seeing what we'd specifically have to keep and what we can get away with, and then I can cut down on the copied code once it's running. But I'm pretty sure that only the two aforementioned things are really necessary for our purposes as you mentioned.